### PR TITLE
WIP:Bugfix/#560 add tooltips

### LINKED ISSUE #560: Add Tooltips to buttons
# User Story

### As a: 
student / teacher

### I want:
I like to see on the first view on the button, which function the button has

## Acceptance criteria:
- [x] add a meaningful term to the buttons
- [x] add term to the language resource file
- [x] translated into the different languages

## Additional info:

------
_Please tag this issue if you are sure to which tag(s) it belongs._

### PR DESCRIPTION
**Pull request by mleppla**
## Description:
- added more tooltips
- add some translations to resource files
- add translations to components

Closes #560 #549

## Improvements
- interface in german and english language
- more information for users when hover buttons

## Known Issues:
_NONE_

### CODE REVIEW

**Comment by PatrickSkowronek** [comment]
@mleppla @wurmc you have multiple error's in the frontend build, with  `--prod`

**Comment by mleppla** [comment]
@PatrickSkowronek fixed, but waiting for coverage

File: app/webFrontend/src/app/lecture/lecture-form/lecture-form.component.html [hunk 1]
```diff
@@ -1,14 +1,14 @@
 <form [formGroup]="lectureForm" (ngSubmit)="formSubmit()">
   <mat-form-field>
-    <input matInput formControlName="name" placeholder="Name">
+    <input matInput formControlName="name" [placeholder]="'common.name' | translate">
```
Comment (PatrickSkowronek) [inline app/webFrontend/src/app/lecture/lecture-form/lecture-form.component.html]: The translate is missing in the component.ts!

File: app/webFrontend/src/app/lecture/lecture-form/lecture-form.component.html [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
   </mat-form-field>
 
   <mat-form-field>
-    <textarea matInput formControlName="description" placeholder="Description"></textarea>
+    <textarea matInput formControlName="description" [placeholder]="'common.description' | translate"></textarea>
```
Comment (PatrickSkowronek) [inline app/webFrontend/src/app/lecture/lecture-form/lecture-form.component.html]: The translate is missing in the component.ts!

File: app/webFrontend/src/app/lecture/lecture-form/lecture-form.component.html [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
   </mat-form-field>
 
   <p>
     <button mat-raised-button [disabled]="!lectureForm.valid" color="primary" type="submit">{{ actionLabel }}</button>
-    <button mat-raised-button (click)="onCancel()">Cancel</button>
+    <button mat-raised-button (click)="onCancel()">{{ 'common.cancel' | translate }}</button>
```
Comment (PatrickSkowronek) [inline app/webFrontend/src/app/lecture/lecture-form/lecture-form.component.html]: The translate is missing in the component.ts!

File: app/webFrontend/src/app/lecture/lecture-form/lecture-form.component.html (hunk 1 above)
Comment (mleppla) [reply app/webFrontend/src/app/lecture/lecture-form/lecture-form.component.html]: ok.. in travis there aren't any errors. I can not test 30 HTML-pages... I have a look on this again this evening or tomorrow.

File: app/webFrontend/src/app/course/course-container/course-container.component.html [hunk 4]
```diff
@@ -2,15 +2,15 @@
 
   <app-badge [displayNumber]="courses.length"></app-badge>
 
-  <button class="open-toggle" [class.opened]="expand" mat-icon-button>
+  <button class="open-toggle" [matTooltip]="'common.tooltip.clickToOpen' | translate" [class.opened]="expand" mat-icon-button>
```
Comment (PatrickSkowronek) [inline app/webFrontend/src/app/course/course-container/course-container.component.html]: I think we don't need this tooltip

Comment (mleppla) [reply app/webFrontend/src/app/course/course-container/course-container.component.html]: deleted

**Pull request merged**